### PR TITLE
Upgrade to cert-manager 1.11

### PR DESCRIPTION
**Pull request by wallrj**
This is in preparation for upgrading to cert-manager 1.12 where we introduce the extra go.mod files.
I want to see if those separate modules results in fewer transitive dependencies in  this sample-external-issuer...

* Upgraded the cert-manager Go dependency  to 1.11
* Use cert-manager 1.11 in E2E tests
* Use k8s.io/utils module

Ran E2E tests locally:
```console
$ make docker-build kind-load deploy e2e
...
Successfully tagged ghcr.io/cert-manager/sample-external-issuer/controller:v0.2.0-alpha.0-18-gab493c2
/home/richard/projects/cert-manager/sample-external-issuer/bin/kind-0.12.0 load docker-image --name sample-external-issuer-e2e ghcr.io/cert-manager/sample-external-issuer/controller:v0.2.0-alpha.0-18-gab493c2
Image: "ghcr.io/cert-manager/sample-external-issuer/controller:v0.2.0-alpha.0-18-gab493c2" with ID "sha256:c8c1cba57ffceb3c293014315611790007ca2a24fa83d494d3639ab679e2eeeb" not yet present on node "sample-external-issuer-e2e-control-plane", loading...
...
deployment.apps/sample-external-issuer-controller-manager configured
...
issuer.sample-issuer.example.com/issuer-sample condition met
certificaterequest.cert-manager.io/issuer-sample condition met
certificate.cert-manager.io/certificate-by-issuer condition met
clusterissuer.sample-issuer.example.com/clusterissuer-sample condition met
certificaterequest.cert-manager.io/clusterissuer-sample condition met
certificate.cert-manager.io/certificate-by-clusterissuer condition met
...
secret "issuer-sample-credentials" deleted
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -439,6 +439,18 @@ kubectl wait --for=condition=Ready --timeout=5s  certificates.cert-manager.io ce
...
+```bash
+# Create a Kind cluster along with cert-manager.
+make kind-cluster deploy-cert-manager
+
+# Wait for cert-manager to start...
+
+# Build and install sample-external-issuer and run the E2E tests.
+# This step can be run iteratively when ever you make changes to the code or to the installation manifests.
+make docker-build kind-load deploy e2e
+```
```
Comment (wallrj) [inline README.md:453]: I've added this documentation for now.
I know I should add GitHub actions workflow for this, but that will be in another PR.

File: go.mod [hunk 2]
```diff
@@ -3,65 +3,77 @@ module github.com/cert-manager/sample-external-issuer
...
 
 require (
-	github.com/cert-manager/cert-manager v1.8.0
-	github.com/go-logr/logr v1.2.0
-	github.com/stretchr/testify v1.7.0
-	k8s.io/api v0.23.4
-	k8s.io/apimachinery v0.23.4
-	k8s.io/client-go v0.23.4
-	sigs.k8s.io/controller-runtime v0.11.1
+	github.com/cert-manager/cert-manager v1.11.1
```
Comment (wallrj) [inline go.mod:6]: This is the only line I changed manually.
The rest was changed by `go mod tidy`.

File: internal/controllers/certificaterequest_controller.go [hunk 3]
```diff
@@ -28,8 +28,8 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/apimachinery/pkg/util/clock"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/utils/clock"
```
Comment (wallrj) [inline internal/controllers/certificaterequest_controller.go:32]: This module got spun out some time ago. 
Upgrading cert-manager brings a newer version of apimachinery which no longer has the clock package.

File: README.md [hunk 4]
```diff
@@ -439,6 +439,18 @@ kubectl wait --for=condition=Ready --timeout=5s  certificates.cert-manager.io ce
 You can of course write more complete tests than this,
 but this is a good start and demonstrates that the issuer is doing what we hoped it would do.
 
+Run the tests as follows:
+
+```bash
+# Create a Kind cluster along with cert-manager.
+make kind-cluster deploy-cert-manager
```
Comment (irbekrm) [inline README.md:446]: Does this work? I think the cert-manager created via `make deploy-cert-manager` also requires gateway-api to be installed

Comment (wallrj) [reply README.md:446]: It does work...perhaps because the tests still install K8S 1.23?

```
$ cmctl version --short
Client Version: v1.11.0
Server Version: v1.11.1

$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.3
Kustomize Version: v4.5.7
Server Version: v1.23.4
WARNING: version difference between client (1.25) and server (1.23) exceeds the supported minor version skew of +/-1

$ make e2e
kubectl apply --filename config/samples
kubectl wait --for=condition=Ready --timeout=5s issuers.sample-issuer.example.com issuer-sample
kubectl wait --for=condition=Ready --timeout=5s  certificaterequests.cert-manager.io issuer-sample
kubectl wait --for=condition=Ready --timeout=5s  certificates.cert-manager.io certificate-by-issuer
kubectl wait --for=condition=Ready --timeout=5s clusterissuers.sample-issuer.example.com clusterissuer-sample
kubectl wait --for=condition=Ready --timeout=5s  certificaterequests.cert-manager.io clusterissuer-sample
kubectl wait --for=condition=Ready --timeout=5s  certificates.cert-manager.io certificate-by-clusterissuer
kubectl delete --filename config/samples
certificate.cert-manager.io/certificate-by-clusterissuer created
certificate.cert-manager.io/certificate-by-issuer created
certificaterequest.cert-manager.io/clusterissuer-sample created
certificaterequest.cert-manager.io/issuer-sample created
clusterissuer.sample-issuer.example.com/clusterissuer-sample created
issuer.sample-issuer.example.com/issuer-sample created
secret/clusterissuer-sample-credentials created
secret/issuer-sample-credentials created
issuer.sample-issuer.example.com/issuer-sample condition met
certificaterequest.cert-manager.io/issuer-sample condition met
certificate.cert-manager.io/certificate-by-issuer condition met
clusterissuer.sample-issuer.example.com/clusterissuer-sample condition met
certificaterequest.cert-manager.io/clusterissuer-sample condition met
certificate.cert-manager.io/certificate-by-clusterissuer condition met
certificate.cert-manager.io "certificate-by-clusterissuer" deleted
certificate.cert-manager.io "certificate-by-issuer" deleted
certificaterequest.cert-manager.io "clusterissuer-sample" deleted
certificaterequest.cert-manager.io "issuer-sample" deleted
clusterissuer.sample-issuer.example.com "clusterissuer-sample" deleted
issuer.sample-issuer.example.com "issuer-sample" deleted
secret "clusterissuer-sample-credentials" deleted
secret "issuer-sample-credentials" deleted
```

Comment (irbekrm) [reply README.md:446]: yeah I didn't realize this repo had make targets to deploy cert-manager 👍🏼

**Review by irbekrm — APPROVED** [review]
Thank you for leaving the code comments 👍🏼 

/lgtm
